sdk-base/src/main/java/com/mapbox/maps/plugin/viewport/experimental/ViewportStatus.kt

```diff
@@ -0,0 +1,31 @@
+package com.mapbox.maps.plugin.viewport.experimental
+
+import com.mapbox.maps.plugin.viewport.experimental.state.ViewportState
+import com.mapbox.maps.plugin.viewport.experimental.transition.ViewportTransition
+
+/**
+ * Represents the status of the viewport.
+ *
+ * It could be either a [ViewportState] or [ViewportTransition].
+ */
+sealed class ViewportStatus {
+  /**
+   * Represents the current status is a [ViewportState].
+   */
+  class State(val state: ViewportState) : ViewportStatus() {
+    override fun getTargetState(): ViewportState {
+      return state
+    }
+  }
+
+  /**
+   * Represents the current status is a [ViewportTransition].
+   */
+  class Transition(val transition: ViewportTransition, val fromState: ViewportState?, val toState: ViewportState) : ViewportStatus() {
+    override fun getTargetState(): ViewportState {
+      return toState
+    }
+  }
+
+  abstract fun getTargetState(): ViewportState
```

Comment:
I don't think I'd include this in the public API.

Reply:
I find this API to be very easy to use when user want to know what the next state to be, regardless current status is a state or a transition.

Reply:
@macdrevx How does the API look like on iOS if the user want to know the next state to be? in order to decide which state to transition to?

Reply:
From discussion: we decided to remove this from public API. developers can add an extension method to do this if needed.